src/Microsoft.SqlTools.ServiceLayer/Connection/Contracts/ConnectionDetails.cs

```diff
@@ -150,15 +150,15 @@ public string EnclaveAttestationUrl
         /// Gets or sets a <see cref="string"/> value that indicates encryption mode that SQL Server should use to perform SSL encryption for all the data sent between the client and server. Supported values are: Optional, Mandatory, Strict, True, False, Yes and No.
         /// Boolean 'true' and 'false' will also continue to be supported for backwards compatibility.
         /// </summary>
-        public string Encrypt
+        public string? Encrypt
         {
             get
             {
-                string value = GetOptionValue<string>("encrypt");
+                string? value = GetOptionValue<string>("encrypt");
```

Comment:
`GetOptionValue<string?>`

We should be using the proper nullable types whenever possible for clarity of what to expect.

Reply:
```suggestion
                string? value = GetOptionValue<string?>("encrypt");
```